File: go-leet-code/main/560. Subarray Sum Equals K.go
package main

func subarraySum(nums []int, k int) int {
	hash := make(map[int32]int32, len(nums))
	ret := int32(0)
	hash[0] = 1
	cumulative := 0
	for _, value := range nums {
		cumulative += value
		count := hash[int32(cumulative-k)]
		ret += count

		hash[int32(cumulative)] += 1
	}
	return int(ret)
}

File: Leetcode practice/560. Subarray Sum Equals K.cpp
#include <iostream>
#include <map>
#include <vector>
using namespace std;

class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
         map<int,int> freq;
         freq[0] = 1;
         vector<int> cumulative;
         int total = 0;
         for(int &num:nums) {
             total += num;
             cumulative.push_back(total);
         }
         int ret = 0;
        for(int &num: cumulative) {
            ret += freq[num-k];
            freq[num]++;
        }
        return ret;
    }
};

int main() {

}